precompiles/p256verify/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+// Copyright 2024 Moonbeam Foundation.
+// This file is part of Moonbeam.
+
+// Moonbeam is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Moonbeam is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Moonbeam.  If not, see <http://www.gnu.org/licenses/>.
+
+// “secp256r1” is a specific elliptic curve, also known as “P-256”
+// and “prime256v1” curves.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+extern crate alloc;
+
+use alloc::vec::Vec;
+use core::marker::PhantomData;
+use fp_evm::{
+	ExitError, ExitSucceed, Precompile, PrecompileFailure, PrecompileHandle, PrecompileOutput,
+	PrecompileResult,
+};
+use frame_support::{traits::Get, weights::Weight};
+use p256::ecdsa::{signature::hazmat::PrehashVerifier, Signature, VerifyingKey};
+
+pub struct P256Verify<W: Get<Weight>>(PhantomData<W>);
+
+impl<W: Get<Weight>> P256Verify<W> {
+	/// Expected input length (160 bytes)
+	const INPUT_LENGTH: usize = 160;
+
+	fn handle_cost(handle: &mut impl PrecompileHandle) -> Result<(), ExitError> {
+		let weight = W::get();
+		handle.record_external_cost(Some(weight.ref_time()), Some(weight.proof_size()), None)
```

Comment:
```suggestion
		handle.record_external_cost(Some(weight.ref_time()), None, None)
```

it's a pure function that doesn't read the storage, so we don't need to account for the proof size